.github/workflows/ci.yaml

```
@@ -99,8 +99,16 @@ jobs:
 
   publish:
     name: Publish
-    needs: nx
-    if: ${{ needs.nx.outputs.affected-plugins != '[]' && needs.nx.outputs.affected-plugins != '' }}
+    needs: [nx, plugin-version-validator]
+    if: >-
+      (
+        github.event_name == 'pull_request' && 
+        needs.plugin-version-validator.result == 'success' &&
+        needs.nx.outputs.affected-plugins != '[]' && 
+        needs.nx.outputs.affected-plugins != ''
+      ) || (
+        github.event_name != 'pull_request'
+      )
```

As we are moving to publish on comment, I'd suggest to just check if the trigger is a push event on master

```suggestion
    needs: nx
    if: ${{ github.event_name == 'push' && needs.nx.outputs.affected-plugins != '[]' && needs.nx.outputs.affected-plugins != '' }}
```